services/worker-manager/src/provider.js

```diff
@@ -67,49 +87,46 @@ class Provider {
   async cleanup() {
   }
 
-  /**
-   * Any code which is required to be run by this Provider must only be
-   * initiated by this method.  If there's any taskcluster-lib-iterate loops to
-   * run, this is where they should be initiated.  Once the returned promise is
-   * resolve, the Provider must be fully working.
+  /*
+   * Called before an iteration of the worker scanner
    */
-  async initiate() {
+  async scanPrepare() {
   }
 
-  /**
-   * Terminate any code which was started by .initiate();
+  /*
+   * Called for every worker on a schedule so that we can update the state of
+   * the worker locally
    */
-  async terminate() {
+  async checkWorker({Worker}) {
   }
 
-  /**
-   * Terminate all workers managed by this provider.  This method must not
-   * return until the request to terminate all workers is completed.  It does
-   * not need to wait until all workers actually terminate.
+  /*
+   * Called after an iteration of the worker scanner
    */
-  async terminateAllWorkers() {
-    throw new Error('Method Unimplemented!');
```

Comment:
Should we leave these errors in place? They help making sure all the required methods are implemented when writing another provider

Reply:
They would help but not all providers will implement this endpoint. For instance static workers running in our datacenters don't have anything to do for this. I expect we may need to create an endpoint that the UI can call to ask what actions a provider supports. Does that make sense to you? We can make a bug to track that work if so.